marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -127,78 +133,98 @@ handleAction Init = do
       handleAction $ PickupAction $ Pickup.PickupWallet walletDetails
   void $ subscribe currentSlotSubscription
 
-handleAction (SetNewWalletNickname nickname) = assign _newWalletNickname nickname
+handleAction (SetNewWalletNicknameString walletNicknameString) = assign (_newWalletDetails <<< _walletNicknameString) walletNicknameString
 
-handleAction (SetNewWalletContractId contractId) = do
-  assign _newWalletContractId contractId
-  --TODO: use the contract ID to lookup the wallet's public key from PAB
-  randomNumber <- liftEffect random
-  let
-    pubKey = show randomNumber
-  assign _remoteDataPubKey $ Success pubKey
+handleAction (SetNewWalletContractIdString contractInstanceIdString) = do
+  assign (_newWalletDetails <<< _contractInstanceIdString) contractInstanceIdString
+  for_ (contractInstanceIdFromString contractInstanceIdString)
+    $ \contractInstanceId -> do
+        remoteDataWalletCompanionContract <- getContractInstance contractInstanceId
+        case remoteDataWalletCompanionContract of
+          Success contractInstanceClientState -> do
+            --TODO: use the contract ID to lookup the wallet's ID from PAB
+            assign (_newWalletDetails <<< _remoteDataWallet) $ Success (Wallet $ fromInt 1)
```

Comment:
remember to use a single `modify` with multiple `set` as each `assign` re-renders the whole thing

Reply:
:+1: